python/pipeline/utils/galvo_corrections.py

```diff
@@ -20,32 +22,68 @@ def correct_motion(img, xymotion):
-    """
-    raster correction for resonant scanners.
-    :param img: 5D image [x, y, nchannel, nslice, nframe].
-    :param raster_phase: phase difference beetween odd and even lines.
-    :param fill_fraction: ratio between active acquisition and total length of the scan line. see scanimage.
-    :return: raster-corrected image [x, y, nchannel, nslice, nframe].
+def correct_raster(scan, raster_phase, fill_fraction, in_place=True):
+    """ Raster correction for resonant scanners.
+
+    Corrects EM images in n-dimensional scans, usual shape is [image_height, image_width,
+    channels, slices, num_frames]. Woks for 2-d arrays and up (assuming images are in the
+    first two dimensions). Based on Matlab implementation of ne7.ip.correctRaster()
+
+    :param np.array scan: Volume with images to be corrected in the first two dimensions.
+    :param float raster_phase: Phase difference beetween odd and even lines.
+    :param float fill_fraction: Ratio between active acquisition and total length of the
+    scan line.
+    :param bool in_place: If True (default), the original array is modified in memory.
+    :return: Raster-corrected scan.
+    :rtype: Same as scan.
+    :raises: PipelineException: Scan is not an np.array with at least 2 dimensions.
     """
-    img = np.array(img)
-    assert img.ndim <= 5, 'Image size greater than 5D.'
-    ix = np.arange(-img.shape[1]/2 + 0.5, img.shape[1]/2 + 0.5) / (img.shape[1]/2)
-
-    tx = np.arcsin(ix * fill_fraction)
-    for ichannel in range(img.shape[2]):
-        for islice in range(img.shape[3]):
-            for iframe in range(img.shape[4]):
-                im = img[:, :, ichannel, islice, iframe].copy()
-                extrapval = np.mean(im)
-                img[::2, :, ichannel, islice, iframe] = interp1d(ix, im[::2, :], kind='linear', bounds_error=False,
-                                                                  fill_value=extrapval)(np.sin(tx +
-                                                                                        raster_phase)/fill_fraction)
-
-                img[1::2, :, ichannel, islice, iframe] = interp1d(ix, im[1::2, :], kind='linear', bounds_error=-False,
-                                                                  fill_value=extrapval)(np.sin(tx -
-                                                                                        raster_phase)/fill_fraction)
-    return img
+    # Basic checks
+    if not isinstance(scan, np.ndarray):
+        raise PipelineException('Scan needs to be a numpy array.')
+    if scan.ndim < 2:
+        raise PipelineException('Scan with less than 2 dimensions.')
+    if not np.issubdtype(scan.dtype, np.float):
+        print('Changing scan from', str(scan.dtype), 'to double')
+        scan = np.double(scan)
+
+    # Get some dimensions
+    original_shape = scan.shape
+    image_height = original_shape[0]
+    image_width = original_shape[1]
+    half_width = image_width / 2
+
+    # ??
```

Comment:
comment should say something like "create interpolation points for sinusoidal raster"